src/analyserwaveform.cpp

```diff
@@ -165,16 +165,18 @@ bool AnalyserWaveform::loadStored(TrackPointer tio) const {
     return false;
 }
 
-void AnalyserWaveform::resetFilters(TrackPointer tio, int sampleRate) {
-    Q_UNUSED(tio);
-    Q_UNUSED(sampleRate);
-    //TODO: (vRince) bind this with *actual* filter values ...
+void AnalyserWaveform::createFilters(int sampleRate) {
     // m_filter[Low] = new EngineFilterButterworth8(FILTER_LOWPASS, sampleRate, 200);
     // m_filter[Mid] = new EngineFilterButterworth8(FILTER_BANDPASS, sampleRate, 200, 2000);
     // m_filter[High] = new EngineFilterButterworth8(FILTER_HIGHPASS, sampleRate, 2000);
     m_filter[Low] = new EngineFilterBessel4Low(sampleRate, 600);
     m_filter[Mid] = new EngineFilterBessel4Band(sampleRate, 600, 4000);
     m_filter[High] = new EngineFilterBessel4High(sampleRate, 4000);
+    // settle filters for silence in preroll to avoids ramping (Bug #1406389)
+    CSAMPLE buf = 0.0;
+    for (int i = 0; i < FilterCount; ++i) {
+        m_filter[i]->process(&buf, &buf, 1);
```

Comment:
This relies on the implementation not barfing on the buffer size being even. It'd feel nicer if this was a length 2 buffer. 

Though it also kind of assumes details of the implementation of EngineFilterIIR. It'd be better if there was some method like "clear ramping" so that we are explicit about what we want it to do.